foxai/explainer/computer_vision/algorithm/input_x_gradient.py

```diff
@@ -20,194 +22,150 @@ class BaseInputXGradientSHAPCVExplainer(Explainer):
 
                 Default: None
             additional_forward_args: If the forward function
                 requires additional arguments other than the inputs for
                 which attributions should not be computed, this argument
-                can be provided. It must be either a single additional
-                argument of a Tensor or arbitrary (non-tuple) type or a tuple
+                can be provided. It must be a tuple
                 containing multiple additional arguments including tensors
                 or any arbitrary python types. These arguments are provided to
                 forward_func in order following the arguments in inputs.
                 Note that attributions are not computed with respect
                 to these arguments.
                 Default: None
-            attribute_to_layer_input: Indicates whether to
-                compute the attribution with respect to the layer input
-                or output. If `attribute_to_layer_input` is set to True
-                then the attributions will be computed with respect to
-                layer input, otherwise it will be computed with respect
-                to layer output.
-                Default: False
-            layer: Layer for which attributions are computed.
-                If None provided, last convolutional layer from the model
-                is taken.
-                Default: None
 
         Returns:
             The input x gradient with respect to each input feature or gradient
             and activation for each neuron in given layer output. Attributions
-            will always be the same size as the provided inputs, with each value
-            providing the attribution of the corresponding input index.
-            If a single tensor is provided as inputs, a single tensor is
-            returned. If a tuple is provided for inputs, a tuple of
-            corresponding sized tensors is returned.
+            will always be the same size as the provided input.
 
         Raises:
             RuntimeError: if attribution has shape (0).
         """
-        attributions: AttributionsType
-        input_x_gradient = self.create_explainer(model=model, layer=layer)
-
-        if isinstance(input_x_gradient, LayerGradientXActivation):
-            attributions = input_x_gradient.attribute(
-                input_data,
-                target=pred_label_idx,
-                additional_forward_args=additional_forward_args,
-                attribute_to_layer_input=attribute_to_layer_input,
-            )
-        else:
-            attributions = input_x_gradient.attribute(
-                input_data,
-                target=pred_label_idx,
-                additional_forward_args=additional_forward_args,
-            )
-        validate_result(attributions=attributions)
-        return attributions
 
+        grad_required = input_data.requires_grad
+        input_data.requires_grad_()
```

Comment:
Is it really necessary to have `requires_grad` set in input data to be able to compute gradients for explanations?

Reply:
If not, test_model_explanation_with_context_manager_and_raw test fails, so I assume yes. 

```
> >           return Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
                t_outputs, grad_outputs_, retain_graph, create_graph, t_inputs,
                allow_unused, accumulate_grad=False)  # Calls into the C++ engine to run the backward pass
E           RuntimeError: One of the differentiated Tensors does not require grad
```